bundles/com.espressif.idf.core/src/com/espressif/idf/core/IDFDynamicVariables.java

```diff
@@ -0,0 +1,24 @@
+package com.espressif.idf.core;
+
+/**
+ * Class to contain all the variables that can be resolved by the IDFDynamicVariableResolver.java
+ * @author Ali Azam Rana
+ *
+ */
+public interface IDFDynamicVariables
```

Comment:
in my opinion, using an enum here can increase readability. For example:
public enum IDFDynamicVariables {
IDF_PY,
IDF_PATH,
IDF_PYTHON_ENV_PATH
}

looks much cleaner 
But it is more personal preference, not necessary to change